imsim/lsst_image.py

```diff
@@ -266,9 +268,27 @@ def addNoise(self, image, config, base, image_num, obj_num, current_var, logger)
             logger.info("Applying vignetting according to radial spline model.")
             radii = Vignetting.get_pixel_radii(camera[det_name])
             sky.array[:] *= self.vignetting.apply_to_radii(radii)
-
+            
+        if self.apply_fringing:
+            # get det number
+            det_num = base['output']['det_num']['current'][0]
+            # get center ra/dec
+            cra = config['wcs']['boresight']['cboresight']['_kd'][1]\
```

Comment:
I poked around a bit and found something that's cleaner and probably more robust than these 10 levels of `[...]`s:
```
boresight, _ = galsim.config.ParseValue(config['wcs'], 'boresight', base, galsim.CelestialCoord)
```
I would change the interface to `CCD_Fringing`, so that you call it like this:
```
ccd_fringing = CCD_fringing(img_wcs, boresight, seed=det_num, spatial_vary=True)
```
Then in `fringe_variation_level`, I would do
```
dx = self.boresight.ra.deg - self.img_wcs.center.deg
dy = self.boresight.dec.deg - self.img_wcs.center.deg
```
I'm wondering a bit about the data in `data/skyline_var.pkl`, however.  Does just doing linear interpolation in ra, dec still work reasonably for declinations close 90 degrees?  It seems like simulating sky regions near the poles of the coordinate system would be problematic.

Reply:
This is better, but still not great.  You shouldn't rely on the `config['wcs']` field to have a `boresight` parameter.  That's specific to the BatoidWCS, and is not generic.  If a user uses a different WCS type, this will break.
Rather, you should add a `boresight` parameter to the `image` field, which is required whenever `apply_fringing` is True.  In the default config file, you should set this as `boresight: '$boresight'`, since we have this in our `eval_variables` section, so it can be trivially evaluated.

Reply:
I tried adding `boresight: '$boresight'` to the `image` field in parallel with `apply_fringing` parameter. However, I'm getting the error `galsim.errors.GalSimConfigError: Unexpected attribute boresight found`.

Reply:
I think you need to add
```
'boresight': galsim.CelestialCoord,
```
to the `opt` dictionary [here](https://github.com/LSSTDESC/imSim/blob/30f7396ccceb8309989da3369c7923f55d3f537d/imsim/lsst_image.py#L46).